x/transfermiddleware/keeper/ibc.go

```diff
@@ -0,0 +1,112 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	ibcexported "github.com/cosmos/ibc-go/v7/modules/core/exported"
+	"github.com/notional-labs/banksy/v2/x/transfermiddleware/types"
+)
+
+func (keeper Keeper) hasParachainIBCTokenInfo(ctx sdk.Context, nativeDenom string) bool {
+	store := ctx.KVStore(keeper.storeKey)
+	return store.Has(types.GetKeyKeysParachainIBCTokenInfo(nativeDenom))
+}
+
+func (keeper Keeper) handleOverrideSendPacketTransferLogic(
+	ctx sdk.Context,
+	chanCap *capabilitytypes.Capability,
+	sourcePort string, sourceChannel string,
+	timeoutHeight clienttypes.Height,
+	timeoutTimestamp uint64,
+	data []byte,
+) (sequence uint64, err error) {
+	var fungibleTokenPacketData transfertypes.FungibleTokenPacketData
+
+	err = keeper.cdc.Unmarshal(data, &fungibleTokenPacketData)
+	if err != nil {
+		return 0, err
+	}
+	sender, err := sdk.AccAddressFromBech32(fungibleTokenPacketData.Sender)
+	if err != nil {
+		return 0, err
+	}
+
+	// check if denom in fungibleTokenPacketData is native denom in parachain info and
+	parachainInfo := keeper.GetParachainIBCTokenInfo(ctx, fungibleTokenPacketData.Denom)
+	if parachainInfo.ChannelId != sourceChannel {
+		keeper.SendPacket(ctx, chanCap, sourcePort, sourceChannel, timeoutHeight, timeoutTimestamp, data)
+	}
+
+	// burn native token in escrow address
+	transferAmount, ok := sdk.NewIntFromString(fungibleTokenPacketData.Amount)
+
+	// TODO: remove this panic and replace by err hanlde
+	if !ok {
+		panic("cannot parse string amount")
+	}
+	nativeTransferToken := sdk.NewCoin(fungibleTokenPacketData.Denom, transferAmount)
+	ibcTransferToken := sdk.NewCoin(parachainInfo.IbcDenom, transferAmount)
+
+	// burn native token
+	keeper.bankKeeper.BurnCoins(ctx, transfertypes.ModuleName, sdk.NewCoins(nativeTransferToken))
```

Comment:
The nativeTransferToken should be burned from escrowAddress which is defined by sourcePort and sourceChannel [here](https://github.com/cosmos/ibc-go/blob/8571b04d40e58b70caa940c466055138ab18d926/modules/apps/transfer/keeper/relay.go#LL105C31-L105C31)